app/models/concerns/entity_access.rb

```diff
@@ -89,6 +172,63 @@ def editable_role
     self.class.viewable_permission
   end
 
+  private def viewable_user_group_name
+    "#{name} [viewable]"
+  end
+
+  private def editable_user_group_name
+    "#{name} [editable]"
+  end
+
+  # NOTE: this only needs to be run once, and only if the installation was created
+  # prior to release-151
+  def fix_entity_associations
+    # If we already have a system collection in the new mechanism, do nothing
+    existing_collection = Collection.system.where(source: self).exists?
+    unless existing_collection
+      # Find the last collection that would have been created under the old mechanism
+      collection = Collection.system.where(name: name).
+        order(id: :desc).
+        first_or_create
```

Comment:
Are these normally created on-demand? Do we expect every project group or cohort to have a collection/usergroup? If not maybe we don't need to make new records unless the collection/user group already exists.

Also probably should use bang variant:
```suggestion
        first_or_create!
```